amazon-kinesis-connector-flink/src/main/java/software/amazon/kinesis/connectors/flink/FlinkKinesisConsumer.java

```diff
@@ -462,6 +465,25 @@ public void snapshotState(FunctionSnapshotContext context) throws Exception {
 		}
 	}
 
+	/**
+	 * Creates state serializer for kinesis shard sequence number.
+	 * Using of the explicit state serializer with KryoSerializer is needed because otherwise
+	 * users cannot use 'disableGenericTypes' properties with KinesisConsumer, see FLINK-24943 for details
+	 *
+	 * @return state serializer
+	 */
+	@VisibleForTesting
+	static TupleSerializer<Tuple2<StreamShardMetadata, SequenceNumber>> createStateSerializer(ExecutionConfig executionConfig) {
+		// explicit serializer will keep the compatibility with GenericTypeInformation and allow to disableGenericTypes for users
```

Comment:
Just to double check, this is backwards compatible with existing state?

Reply:
Yes, the idea is to keep state of existing applications compatible with new version. Because if we make SequenceNumber POJO it will use difference serde and won't be able to recover state created by Kryo.